linera-chain/src/chain.rs

```diff
@@ -572,26 +574,58 @@ where
                     .await?;
             }
         }
-        // Process the inbox bundle and update the inbox state.
-        let mut inbox = self.inboxes.try_load_entry_mut(origin).await?;
-        let entry = InboxEntry::new(origin.clone(), &bundle);
-        let skippable = bundle.is_skippable();
-        let newly_added = inbox
-            .add_bundle(bundle)
-            .await
-            .map_err(|error| match error {
-                InboxError::ViewError(error) => ChainError::ViewError(error),
-                error => ChainError::InternalError(format!(
-                    "while processing messages in certified block: {error}"
-                )),
-            })?;
-        if newly_added && !skippable {
-            let seen = local_time;
-            self.unskippable
-                .push_back(TimestampedInboxEntry { entry, seen });
+        if bundle.messages.iter().all(|posted_message| {
+            matches!(
+                posted_message.message,
+                Message::System(
+                    SystemMessage::Subscribe { .. } | SystemMessage::Unsubscribe { .. }
+                )
+            )
+        }) {
```

Comment:
We want to avoid putting all the `Subscribe` messages in the admin chain's inbox. But I can move the check into `linera-execution`, so `SystemMessage` doesn't leak into other crates.